L1Trigger/L1TTrackMatch/plugins/L1TkMuonProducer.cc

```diff
@@ -0,0 +1,712 @@
+
+  std::string bmtfMatchAlgoVersionString = iConfig.getParameter<std::string>("bmtfMatchAlgoVersion");
+  std::transform(bmtfMatchAlgoVersionString.begin(),
+                 bmtfMatchAlgoVersionString.end(),
+                 bmtfMatchAlgoVersionString.begin(),
+                 ::tolower);  // make lowercase
+
+  std::string omtfMatchAlgoVersionString = iConfig.getParameter<std::string>("omtfMatchAlgoVersion");
+  std::transform(omtfMatchAlgoVersionString.begin(),
+                 omtfMatchAlgoVersionString.end(),
+                 omtfMatchAlgoVersionString.begin(),
+                 ::tolower);  // make lowercase
+
+  std::string emtfMatchAlgoVersionString = iConfig.getParameter<std::string>("emtfMatchAlgoVersion");
+  std::transform(emtfMatchAlgoVersionString.begin(),
+                 emtfMatchAlgoVersionString.end(),
+                 emtfMatchAlgoVersionString.begin(),
+                 ::tolower);  // make lowercase
+
+  if (bmtfMatchAlgoVersionString == "tp")
+    bmtfMatchAlgoVersion_ = kTP;
+  else if (bmtfMatchAlgoVersionString == "mantra")
+    bmtfMatchAlgoVersion_ = kMantra;
+  else
+    throw cms::Exception("TkMuAlgoConfig") << "the ID of the BMTF algo matcher passed is invalid\n";
+
+  //
+
+  if (omtfMatchAlgoVersionString == "tp")
+    omtfMatchAlgoVersion_ = kTP;
+  else if (omtfMatchAlgoVersionString == "mantra")
+    omtfMatchAlgoVersion_ = kMantra;
+  else
+    throw cms::Exception("TkMuAlgoConfig") << "the ID of the OMTF algo matcher passed is invalid\n";
+
+  //
+
+  if (emtfMatchAlgoVersionString == "tp")
+    emtfMatchAlgoVersion_ = kTP;
+  else if (emtfMatchAlgoVersionString == "dynamicwindows")
+    emtfMatchAlgoVersion_ = kDynamicWindows;
+  else if (emtfMatchAlgoVersionString == "mantra")
+    emtfMatchAlgoVersion_ = kMantra;
+  else
+    throw cms::Exception("TkMuAlgoConfig") << "the ID of the EMTF algo matcher passed is invalid\n";
+
+  correctGMTPropForTkZ_ = iConfig.getParameter<bool>("correctGMTPropForTkZ");
+
+  use5ParameterFit_ = iConfig.getParameter<bool>("use5ParameterFit");
+  useTPMatchWindows_ = iConfig.getParameter<bool>("useTPMatchWindows");
+  produces<TkMuonCollection>();
+
+  // initializations
+  if (emtfMatchAlgoVersion_ == kDynamicWindows) {
+    // FIXME: to merge eventually into an unique file with bith phi and theta boundaries
+    std::string fIn_bounds_name = iConfig.getParameter<edm::FileInPath>("emtfcorr_boundaries").fullPath();
+    std::string fIn_theta_name = iConfig.getParameter<edm::FileInPath>("emtfcorr_theta_windows").fullPath();
+    std::string fIn_phi_name = iConfig.getParameter<edm::FileInPath>("emtfcorr_phi_windows").fullPath();
+    auto bounds = L1TkMuCorrDynamicWindows::prepare_corr_bounds(fIn_bounds_name, "h_dphi_l");
+    TFile* fIn_theta = TFile::Open(fIn_theta_name.c_str());
+    TFile* fIn_phi = TFile::Open(fIn_phi_name.c_str());
+    dwcorr_ = std::unique_ptr<L1TkMuCorrDynamicWindows>(new L1TkMuCorrDynamicWindows(bounds, fIn_theta, fIn_phi));
```

Comment:
use make_unique (if this needs to be a pointer at all)

Reply:
done